packages/auth-passport/authService.js

```diff
@@ -3,73 +3,56 @@ var CONSTANTS = require('./constants');
 var $fh = require('fh-js-sdk');
 var cloudUrl;
 var userProfile;
-var USER_CACHE_KEY = 'rcuser_profile';
 
-function PassportAuthService($http, $window, $mdDialog) {
+var PassportAuthService = function PassportAuthService($http, $window, $mdDialog, $state) {
   this.http = $http;
   this.window = $window;
   this.dialog = $mdDialog;
+  this.state = $state;
   this.init();
 }
 
 /**
  * Initializes $fh and retrieves the server's URL.
  */
 PassportAuthService.prototype.init = function() {
+  var self = this;
   $fh.on('fhinit', function(error) {
     if (error) {
       logger.error('Unable to initialize auth service due to unsuccessful $fh.init', error);
     } else {
-      cloudUrl = decodeURIComponent($fh.getCloudURL());
+      self.setCloudUrl();
     }
   });
 }
 
 /**
+ * Sets the cloud URL
+ */
+PassportAuthService.prototype.setCloudUrl = function() {
+  // Note: decodeURIComponent is used for backwards compatability from Keycloak to Passport  
+  cloudUrl = decodeURIComponent($fh.getCloudURL());
```

Comment:
cloudUrl then should be part of the class. If we create multiple objects they will all use the same url.